### PR TITLE
[usm] service discovery, simplify function envs.Variables.Get()

### PR DESCRIPTION
**Pull request by yuri-lipnesh**
### What does this PR do?

moves input verification in target map from function envs.Variables.Get() to function envs.NewVariables()

### Motivation
This is a follow-up of another [PR](https://github.com/DataDog/datadog-agent/pull/29574)

### CODE REVIEW

File: pkg/collector/corechecks/servicediscovery/envs/envs.go [hunk 1]
```diff
@@ -35,17 +35,21 @@ type Variables struct {
...
 // NewVariables returns a new [Variables] to collect env. variables.
 func NewVariables(vars map[string]string) Variables {
+	if len(vars) > 0 {
+		// Only some tests pass a non-empty parameter. Check that input variables match the target value.
+		for env := range vars {
+			if _, ok := targets[env]; !ok {
+				return Variables{}
+			}
+		}
+	}
```
Comment (guyarb) [inline pkg/collector/corechecks/servicediscovery/envs/envs.go:45]: 1. iterating over an empty collection is not executing anything, so you can remove the first condition
2. The current code rejects any input that contains at least one unsupported environment variable, that's not the equivalent of the change. You should have an insertion filter to store only supported environment variables

File: pkg/collector/corechecks/servicediscovery/envs/envs.go [hunk 2]
```diff
@@ -35,17 +35,20 @@ type Variables struct {
 
 // NewVariables returns a new [Variables] to collect env. variables.
 func NewVariables(vars map[string]string) Variables {
+	// A regular flow provides an empty input map. For unit tests, verify that the input map has only target variables.
+	for env := range vars {
+		if _, ok := targets[env]; !ok {
+			return Variables{}
+		}
+	}
```
Comment (guyarb) [inline pkg/collector/corechecks/servicediscovery/envs/envs.go:43]: a general comment - 

if we have code piece that is relevant only for testing - it should be separated from the "normal" execution flow, in a dedicated function and behind a `test` buildflag

File: pkg/collector/corechecks/servicediscovery/envs/envs.go (hunk 1 above)
Comment (yuri-lipnesh) [reply pkg/collector/corechecks/servicediscovery/envs/envs.go]: on 1, I removed length condition
on 2, only unit tests call this function with non-empty map, it is enough to have one non-targeted variable in input to fail the unit test. Regular envs reading sets input map empty and fills it reading /proc

File: pkg/collector/corechecks/servicediscovery/envs/envs.go (hunk 2 above)
Comment (yuri-lipnesh) [reply pkg/collector/corechecks/servicediscovery/envs/envs.go]: I've changed in last commit only unit tests call NewVariables()

File: pkg/collector/corechecks/servicediscovery/envs/envs.go [hunk 3]
```diff
@@ -33,19 +33,8 @@ type Variables struct {
 	Vars map[string]string
```
Comment (guyarb) [inline pkg/collector/corechecks/servicediscovery/envs/envs.go:33]: Taking another pass on the code - nothing is accessing `Vars` directly, and instead using `Get` or `Set`
In such a case, let's un-expose this field

**Review by guyarb — APPROVED** [review]
Approved with one suggestion

File: pkg/collector/corechecks/servicediscovery/envs/envs.go (hunk 3 above)
Comment (yuri-lipnesh) [reply pkg/collector/corechecks/servicediscovery/envs/envs.go]: done

**Pull request merged**